### PR TITLE
Support add custom label on GCP compute engine vm

### LINKED ISSUE #82: Support add custom label on GCP compute engine vm
**What would you like to be added**:
In pkg/controller/worker/machines.go,  the machineClassSpec  labels map only contains one key named "name",  can we all pool labels to the machineClassSpec?  
```
	machineClassSpec := map[string]interface{}{
				"region":             w.worker.Spec.Region,
				"zone":               zone,
				"canIpForward":       true,
				"deletionProtection": false,
				"description":        fmt.Sprintf("Machine of Shoot %s created by machine-controller-manager.", w.worker.Name),
				"disks":              disks,
				"labels": map[string]interface{}{
					"name": w.worker.Name,
				},
```
By the way, AWS extension needs this feature too.

**Why is this needed**:
We need some custom labels to distinguish  VM, then we can charge our tenant by query the cost from the big query.

If it's possible, I can submit a PR to do this

### PR DESCRIPTION
**Pull request by sdojjy**
**What this PR does / why we need it**:
This PR will add worker pool labels to GCP VM, with this feature we can add custom labels to GCP VM

**Which issue(s) this PR fixes**:
Fixes #82

**Special notes for your reviewer**:

**Release note**:

```improvement operator
WorkerPool labels will be added to GCP VM as the resource label
```

### CODE REVIEW

File: pkg/controller/worker/machines.go [hunk 1]
```diff
@@ -147,6 +147,13 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 			disks = append(disks, disk)
 		}
 
+		// gcp vm labels
+		machineClassLabels := map[string]interface{}{
```
Comment (rfranzke) [inline pkg/controller/worker/machines.go:151]:
```suggestion
		gceInstanceLabels := map[string]interface{}{
```

File: pkg/controller/worker/machines.go [hunk 2]
```diff
@@ -147,6 +147,13 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 			disks = append(disks, disk)
 		}
 
+		// gcp vm labels
+		gceInstanceLabels := map[string]interface{}{
+			"name": w.worker.Name,
+		}
+		for k, v := range pool.Labels {
+			machineClassLabels[k] = v
```
Comment (rfranzke) [inline pkg/controller/worker/machines.go:155]:
```suggestion
			gceInstanceLabels[k] = v
```

File: pkg/controller/worker/machines.go [hunk 3]
```diff
@@ -156,10 +163,8 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 				"deletionProtection": false,
 				"description":        fmt.Sprintf("Machine of Shoot %s created by machine-controller-manager.", w.worker.Name),
 				"disks":              disks,
-				"labels": map[string]interface{}{
-					"name": w.worker.Name,
-				},
-				"machineType": pool.MachineType,
+				"labels":             machineClassLabels,
```
Comment (rfranzke) [inline pkg/controller/worker/machines.go:166]:
```suggestion
				"labels":             gceInstanceLabels,
```

**Comment by sdojjy** [comment]
> Can you adapt the unit test accordingly?

OK

**Comment by rfranzke** [comment]
Looks good, thanks! Can you squash the commits into a single one so that we can go ahead with the merge? :)

**Comment by rfranzke** [comment]
Thanks! Looks good, need to play with it - will come back

**Comment by sdojjy** [comment]
> I tried it out, but I ran into a problem. When using these labels
> 
> ```yaml
>   labels:
>     bar: baz-foo_bar.foobar
>     foo: bar
> ```
> 
> then the GCE API rejects the VM creation:
> 
> ```
> E0507 04:43:37.460324       1 machine.go:408] Error while creating machine shoot--proj--shootcpu-worker-z1-7bdb7b8768-qh74k: googleapi: Error 400: Invalid value for field 'resource.labels': ''. Label value 'baz-foo_bar.foobar' violates format constraints. The value can only contain lowercase letters, numeric characters, underscores and dashes. The value can be at most 63 characters long. International characters are allowed., invalid
> ```
> 
> I think you need to sanitize the labels before putting them into the machine class.

thanks, I will update this PR

File: pkg/controller/worker/machines.go [hunk 4]
```diff
@@ -247,3 +260,12 @@ func createDiskSpec(volume v1alpha1.Volume, workerName string, machineImage stri
 
 	return disk, nil
 }
+
+// SanitizeGcpLabel will sanitize the label/value base on the gcp label Restrictions
+func SanitizeGcpLabel(label string) string {
```
Comment (rfranzke) [inline pkg/controller/worker/machines.go:265]: Thank you! Looks good, just a unit test for this function is missing. :)

Comment (sdojjy) [reply pkg/controller/worker/machines.go:271]: good suggestion. done

**Comment by rfranzke** [comment]
@sdojjy I think you have to rebase, sorry, could you please do so that CI can pass?

**Comment by sdojjy** [comment]
> @sdojjy I think you have to rebase, sorry, could you please do so that CI can pass?

OK

**Comment by sdojjy** [comment]
CI will take more than 3 hours?

**Pull request merged**